src/custom-elements/src/components/date/date.tsx

```diff
@@ -0,0 +1,269 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Method,
+  Prop,
+  State,
+  Watch
+} from '@stencil/core';
+import dayjs, { Dayjs } from 'dayjs';
+import { CLASSES } from '../../constants/constants';
+
+const WEEK_CLASS_PART = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat'];
+
+@Component({
+  tag: 'chi-date',
+  styleUrl: 'date.scss',
+  scoped: true
+})
+export class Date {
+  /**
+   * Selected date in the date picker
+   */
+  @Prop({ reflectToAttr: true, mutable: true }) value: string;
+
+  /**
+   * Locale to use in date picker
+   */
+  @Prop({ reflectToAttr: true }) locale = 'en';
+
+  /**
+   * Minimum eligible date
+   */
+  @Prop({ reflectToAttr: true }) min = '01/01/1900';
+
+  /**
+   * Maximum eligible date
+   */
+  @Prop({ reflectToAttr: true }) max = '12/31/2099';
+
+  /**
+   * Date format used in the attributes and how it will be shown to the user.
+   */
+  @Prop({ reflectToAttr: true }) format = 'MM/DD/YYYY';
+
+  @State() viewMonth = dayjs();
+
+  @Watch('value')
+  dateChanged(newValue: string, oldValue: string) {
+    if (newValue !== oldValue) {
+      this._vm.date = newValue ? dayjs(newValue) : null;
+      if (this._vm.date && !this._vm.date.isValid()) {
+        throw new Error(`Date ${newValue} has an invalid format. `);
+      }
+      this.viewMonth = this._vm.date;
+    }
+  }
+
+  @Watch('locale')
+  localeChanged() {
+    this._initCalendarViewModel();
+  }
+
+  @Watch('viewMonth')
+  viewMonthChanged(newValue: Dayjs, oldValue: Dayjs) {
+    if (
+      newValue.month() !== oldValue.month() ||
+      newValue.year() !== oldValue.year()
+    ) {
+      this._updateViewMonth();
+    }
+  }
+
+  @Watch('min')
+  minChanged(newValue: string) {
+    const newMinDate = this.fromString(newValue);
```

Comment:
Please, always leave a blank line after const (and or let) definitions. Seek for further occurrences.

Reply:
@SergioQCL I'll keep it this way as it is formatted with prettier. If you think we should keep a blank line there, then we should change prettier rules.

Reply:
Ok, I agree to let prettier decide :)